aws_lambda_powertools/utilities/data_classes/appsync_resolver_event.py

```diff
@@ -184,71 +183,32 @@ def identity(self) -> Union[None, AppSyncIdentityIAM, AppSyncIdentityCognito]:
         return get_identity_object(self.get("identity"))
 
     @property
-    def source(self) -> Optional[Dict[str, Any]]:
+    def source(self) -> Dict[str, Any]:
         """A map that contains the resolution of the parent field."""
-        return self.get("source")
+        return self.get("source") or {}
 
     @property
-    def request_headers(self) -> Dict[str, str]:
+    def request_headers(self) -> MutableMapping[str, str]:
         """Request headers"""
-        return self["request"]["headers"]
+        return CaseInsensitiveDict(self["request"]["headers"])
 
     @property
-    def prev_result(self) -> Optional[Dict[str, Any]]:
+    def prev_result(self) -> Dict[str, Any]:
         """It represents the result of whatever previous operation was executed in a pipeline resolver."""
         prev = self.get("prev")
         if not prev:
-            return None
-        return prev.get("result")
+            return {}
+        return prev.get("result") or {}
```

Comment:
This might be the only case where returning `Optional[Dict[str, Any]]` could make sense to distinguish between "no previous operation result" (`None`) and "previous operation result is empty" (`{}`). Not sure how the AppSync Resolver Events data looks like.